app/src/ui/history/compare.tsx

```diff
@@ -513,6 +517,29 @@ export class CompareSidebar extends React.Component<
   private onCherryPick = (commits: ReadonlyArray<CommitOneLine>) => {
     this.props.onCherryPick(this.props.repository, commits)
   }
+
+  /**
+   * This method is a generic event handler for when a commit has started being
+   * dragged.
+   *
+   * Currently only used for cherry picking, but this could be more generic.
+   */
+  private onDragCommitStart = (commits: ReadonlyArray<CommitOneLine>) => {
+    this.props.dispatcher.setCherryPickFlowStep(this.props.repository, {
+      kind: CherryPickStepKind.CommitsChosen,
+      commits,
+    })
+  }
+
+  /**
+   * This method is a generic event handler for when a commit has ended being
+   * dragged.
+   *
+   * Currently only used for cherry picking, but this could be more generic.
+   */
+  private onDragCommitEnd = () => {
+    this.props.dispatcher.closeFoldout(FoldoutType.Branch)
```

Comment:
It feels kind of weird closing the foldout from here. But I can't suggest anything better… I think this component has enough context about what's happening.

Reply:
Yeah. I agree - I debated bubbling all the way up to the app component where this sort of action would feel more at home.. but, I don't think we can much from that and the dispatcher was already down at this level.